app/metal-controller-manager/api/v1alpha1/server_types.go

```diff
@@ -34,11 +36,41 @@ type SystemInformation struct {
 	Family       string `json:"family,omitempty"`
 }
 
+func (a *SystemInformation) PartialEqual(b *SystemInformation) bool {
+	return partialEqual(a, b)
+}
+
 type CPUInformation struct {
 	Manufacturer string `json:"manufacturer,omitempty"`
 	Version      string `json:"version,omitempty"`
 }
 
+func (a *CPUInformation) PartialEqual(b *CPUInformation) bool {
+	return partialEqual(a, b)
+}
+
+func partialEqual(a, b interface{}) bool {
```

Comment:
I am sure this function can be picked apart if we wanted to make it super generic. I could spend the time on that, or we could just use it knowing that it only works for structs with a flat list of string fields.